### PR TITLE
SQL queries by Foxtra

### PR DESCRIPTION
**Pull request by Foxtra**
Hello,

please check my pull request for homework 4.

Regards,
Sergei

### CODE REVIEW

**Comment by Foxtra** [comment]
Link to task: #141

File: HW4/Sergey_Tkachev/MyDatabase/Northwind/dbo/Stored Procedures/ProdceduresByFoxtra.sql [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+  from [dbo].[Orders]
+  where year([OrderDate]) = 1998
+  group by  [Orders].[EmployeeID], [Orders].[CustomerID]
+  order by Amount desc;
+
+--2.2.4
+	Select  
+		[Employees].[LastName] + ' ' + [Employees].[FirstName] as Seller
+		,[Customers].[ContactName]
+		,[Employees].[City]
```
Comment (MilRoad) [inline HW4/Sergey_Tkachev/MyDatabase/Northwind/dbo/Stored Procedures/ProdceduresByFoxtra.sql]: We don't need to select column City and there is better to use UNION operator

File: HW4/Sergey_Tkachev/MyDatabase/Northwind/dbo/Stored Procedures/ProdceduresByFoxtra.sql [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+	FROM [dbo].[Employees], [dbo].[Customers]
+	where  [Customers].[City] = [Employees].[City];
+	
+--2.2.5
+	Select  
+		[ContactName]
+		,[City]
+	FROM [dbo].[Customers]
+	group by [City], [ContactName];
```
Comment (MilRoad) [inline HW4/Sergey_Tkachev/MyDatabase/Northwind/dbo/Stored Procedures/ProdceduresByFoxtra.sql]: You just grouped it and didn't make the task

File: HW4/Sergey_Tkachev/MyDatabase/Northwind/dbo/Stored Procedures/ProdceduresByFoxtra.sql [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+	on e.[City] = t.[TerritoryDescription]
+	inner join [dbo].Region as r
+	on t.[RegionID] = r.[RegionID];
+
+--2.3.2
+	Select c.[ContactName], SUM(o.OrderID) as 'Amount of orders'
+	FROM [dbo].[Customers] c
+	left join [dbo].[Orders] o
+	on c.[CustomerID] = o.[CustomerID]
+	group by c.[ContactName];
```
Comment (MilRoad) [inline HW4/Sergey_Tkachev/MyDatabase/Northwind/dbo/Stored Procedures/ProdceduresByFoxtra.sql]: You forgot to order it

File: HW4/Sergey_Tkachev/MyDatabase/Northwind/dbo/Stored Procedures/ProdceduresByFoxtra.sql [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+		e.[LastName] + ' ' + e.[FirstName] as Employee
+		,r.[RegionDescription]
+	FROM [dbo].[Employees] as e
+	inner join [dbo].[Territories] as t
+	on e.[City] = t.[TerritoryDescription]
+	inner join [dbo].Region as r
+	on t.[RegionID] = r.[RegionID];
+
+--2.3.2
+	Select c.[ContactName], SUM(o.OrderID) as 'Amount of orders'
```
Comment (MilRoad) [inline HW4/Sergey_Tkachev/MyDatabase/Northwind/dbo/Stored Procedures/ProdceduresByFoxtra.sql]: You need to use here COUNT, not SUM

File: HW4/Sergey_Tkachev/MyDatabase/Northwind/dbo/Stored Procedures/ProdceduresByFoxtra.sql [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+		  where  e1.[ReportsTo] = e2.[EmployeeID]) as Manager
+	FROM [dbo].[Employees] as e1;
+	
+--2.3.1
+	Select  
+		e.[LastName] + ' ' + e.[FirstName] as Employee
+		,r.[RegionDescription]
+	FROM [dbo].[Employees] as e
+	inner join [dbo].[Territories] as t
+	on e.[City] = t.[TerritoryDescription]
```
Comment (MilRoad) [inline HW4/Sergey_Tkachev/MyDatabase/Northwind/dbo/Stored Procedures/ProdceduresByFoxtra.sql]: I don't think that City column in Employees table is where employees serve. That kind of information is in EmployeeTerritories. That's why it doesn't work right.

**Comment by Foxtra** [comment]
@MilRoad thanks for your comments, I've made the fixes

**Pull request merged**